scripts/imgtool/keys/ecdsa.py

```diff
@@ -140,7 +146,10 @@ def raw_sign(self, payload):
                 signature_algorithm=ec.ECDSA(SHA256()))
 
     def sign(self, payload):
-        # To make fixed length, pad with one or two zeros.
         sig = self.raw_sign(payload)
-        sig += b'\000' * (self.sig_len() - len(sig))
-        return sig
+        if self.pad_sig:
+            # To make fixed length, pad with one or two zeros.
+            sig += b'\000' * (self.sig_len() - len(sig))
+            return sig
+        else:
+            return sig
```

Comment:
I think you could just dedent this `return` and remove the `else` and `return` above.